StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1013CSharp8UnitTests.cs

```diff
@@ -62,5 +74,54 @@ public void TestMethod(object value)
             var expected = Diagnostic().WithSpan(14, 37, 14, 38).WithArguments(string.Empty, "followed");
             await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Validates that a closing brace followed by a null-forgiving operator does not require a space.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [WorkItem(3172, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3172")]
+        public async Task TestCloseBraceWithNullForgivingOperatorAsync()
+        {
+            const string testCode = @"
+public class Foo
+{
+    public void TestMethod()
+    {
+        var test = new[]
+        {
+            new { Value = default(string) },
+            new { Value = ""a"" }!,
+            new { Value = ""b"" } !,
+        };
+    }
+}
+";
+
+            var test = new NullableCSharpAnalyzerTest
+            {
+                TestCode = testCode,
+            };
+
+            await test.RunAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        private class NullableCSharpAnalyzerTest : AnalyzerTest<XUnitVerifier>
+        {
+            public override string Language => LanguageNames.CSharp;
+
+            protected override string DefaultFileExt => "cs";
+
+            protected override CompilationOptions CreateCompilationOptions()
+                => new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe: true, nullableContextOptions: NullableContextOptions.Enable);
```

Comment:
❔ Why not just put `#nullable enable` at the top of the file?

Reply:
Lack of knowledge :wink: I'll change it.